main.py

```diff
@@ -1,144 +1,197 @@
+SORTABLE_FIELDS = Literal[
+    "pokemon_id", "name", "type_1", "type_2", "total",
+    "hp", "attack", "defense", "sp_atk", "sp_def", "speed", "generation", "legendary"
+]
+
+
+@app.post("/pokemon/load")
+def fetch_and_store(db: Session = Depends(get_db), role: str = Depends(role_required(["admin"]))):
+    # Fetch the JSON data from the URL
+    response = requests.get("https://coralvanda.github.io/pokemon_data.json")
+    data = response.json()
+
+    print(f"Data fetched: {len(data)} entries")
+
+    # Collect all existing Pokémon numbers from the database
+    existing_numbers = {pokemon.number for pokemon in db.query(Pokemon.number).all()}
+
+    # Collect Pokémon data for bulk insert
+    pokemon_data = []
+
+    for entry in data:
+        original_number = entry.get("#")
+        number = original_number
+
+        # Find the next available unique number
+        while number in existing_numbers or any(p['number'] == number for p in pokemon_data):
+            number += 1
+        
+        # If the number was incremented, log the change
+        if number != original_number:
+            print(f"Duplicate number {original_number} found. Assigning new number {number}.")
+
+        # Create a dictionary of Pokémon data for bulk insert
+        pokemon_data.append({
+            'number': number,
+            'name': entry.get("Name"),
+            'type_1': entry.get("Type 1"),
+            'type_2': entry.get("Type 2") if entry.get("Type 2") else None,
+            'total': entry.get("Total", 0),
+            'hp': entry.get("HP", 0),
+            'attack': entry.get("Attack", 0),
+            'defense': entry.get("Defense", 0),
+            'sp_atk': entry.get("Sp. Atk", 0),
+            'sp_def': entry.get("Sp. Def", 0),
+            'speed': entry.get("Speed", 0),
+            'generation': entry.get("Generation", 0),
+            'legendary': entry.get("Legendary", False)
+        })
+
+        # Add the new unique number to the set of existing numbers
+        existing_numbers.add(number)
+
+    # Perform bulk insert using bulk_insert_mappings
+    try:
+        db.bulk_insert_mappings(Pokemon, pokemon_data)
+        db.commit()
+        return {
+            "message": "Authorized and Data successfully stored in the database",
+            "inserted": len(pokemon_data),
+        }
+    except SQLAlchemyError as e:
+        db.rollback()  
+        return {"error": str(e)}
+
+
+@app.get("/pokemon/", response_model=PokemonGetAllOutputSchema)
 def read_pokemon(
-    sort_by: str = Query(default="pokemon_id", description="Column to sort by", regex="^(pokemon_id)$"),
-    order: str = Query(default="asc", description="Sort order: 'asc' or 'desc'", regex="^(asc|desc)$"),
-    search_column: str = Query(default="name", description="Column to search in"),
-    keyword: Optional[str] = Query(None, description="Keyword to search for")
+    sort_by: SORTABLE_FIELDS = Query(default="pokemon_id", description="Column to sort by"),
```

Comment:
i understand what you are trying to do here.

but do it this way:
in constants.py create "SORTABLE_FIELDS" value with list of field names

SORTABLE_FIELDS = [ 'field1','field2'....]

and in here
sort_by: Literal[ *SORTABLE_FIELDS ]
this will explain that we are limiting some inputs